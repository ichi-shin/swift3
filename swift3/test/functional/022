#!/bin/bash

# DELETE Multiple Objects

. ./common

num=10

S3USER=tester

_gen_delete_xml()
{
    echo '<Delete>'
    echo '<Quiet>false</Quiet>'

    for i in `seq $num`; do
	echo "<Object><Key>object$i</Key></Object>"
    done

    echo '</Delete>'
}

_gen_delete_xml > $tmp.delete.xml

_s3_put /bucket

for i in `seq $num`; do
    _s3_put /bucket/object$i -T /dev/null
done

MD5=$(_md5 $tmp.delete.xml) _s3_post /bucket?delete -T $tmp.delete.xml -D - | _filter_curl xml

_sw_get /bucket | _filter_curl
