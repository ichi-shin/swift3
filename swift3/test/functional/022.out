QA output created by 022
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object1... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object2... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object3... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object4... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object5... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object6... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object7... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object8... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object9... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object10... 200
> s3curl --id tester --contentMd5 hV8OySdDSDx2uwoLKd3maw== -- -X POST -T /TMP.delete.xml -D - http://SWIFT_HOST/bucket?delete... 200
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<DeleteResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Deleted>
    <Key>object1</Key>
  </Deleted>
  <Deleted>
    <Key>object2</Key>
  </Deleted>
  <Deleted>
    <Key>object3</Key>
  </Deleted>
  <Deleted>
    <Key>object4</Key>
  </Deleted>
  <Deleted>
    <Key>object5</Key>
  </Deleted>
  <Deleted>
    <Key>object6</Key>
  </Deleted>
  <Deleted>
    <Key>object7</Key>
  </Deleted>
  <Deleted>
    <Key>object8</Key>
  </Deleted>
  <Deleted>
    <Key>object9</Key>
  </Deleted>
  <Deleted>
    <Key>object10</Key>
  </Deleted>
</DeleteResult>
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket... 204
